File: 2016/03.pl
#!/usr/bin/env perl
use strict;
use warnings;

sub is_triangle {
  my ($a, $b, $c) = @_;

  if (($a + $b) <= $c) {
    return 0;
  }

  if (($a + $c) <= $b) {
    return 0;
  }

  if (($b + $c) <= $a) {
    return 0;
  }

  return 1;
}

open my $input, "<", "03_input.txt" or die "Can't open input file";

my $count = 0;
while (<$input>) {
  $_ =~ s/^\s+|\s+$//g;
  my @triangle = split " ", $_;
  $count += (is_triangle(@triangle)) ? 1 : 0;
}

print "Answer: $count\n";
close $input;

File: 2018/03.py
#!/usr/bin/env python3
from collections import defaultdict

def parse_line(line):
  args = line.strip().split()
  x, y = args[2].split(',')
  y = y[:-1]
  width, height = args[3].split('x')
  return (int(x), int(y), int(width), int(height))


def update_cells(x, y, width, height, cells):
  for i in range(width):
    for j in range(height):
      cells[(i+x, j+y)] += 1


cells = defaultdict(lambda: 0, defaultdict(int))
with open('03_input.txt', 'r') as data:
  for line in data:
    x, y, width, height = parse_line(line)
    update_cells(x, y, width, height, cells)

answer = 0
for count in cells.values():
  if count >= 2:
    answer += 1
print('Answer:', answer)
